lifecycle/src/lifecycle_service_client.cpp

```diff
@@ -0,0 +1,161 @@
+  {}
+
+  void
+  init()
+  {
+    // Service client pointing to each individual service
+    client_get_state_ = this->create_client<lifecycle_msgs::srv::GetState>(
+      node_get_state_topic);
+    client_change_state_ = this->create_client<lifecycle_msgs::srv::ChangeState>(
+      node_change_state_topic);
+  }
+
+  unsigned int
+  get_state(std::chrono::seconds time_out = 3_s)
+  {
+    auto request = std::make_shared<lifecycle_msgs::srv::GetState::Request>();
+
+    if (!client_get_state_->wait_for_service(time_out)) {
+      fprintf(stderr, "Service %s is not available.\n",
+        client_get_state_->get_service_name().c_str());
+      return lifecycle_msgs::msg::State::PRIMARY_STATE_UNKNOWN;
+    }
+
+    auto result = client_get_state_->async_send_request(request);
+    if (result.wait_for(std::chrono::milliseconds(time_out)) != std::future_status::ready) {
+      fprintf(stderr, "[%s] Failed to get current state for node %s. Server timed out.\n",
+        get_name(), lifecycle_node);
+      return lifecycle_msgs::msg::State::PRIMARY_STATE_UNKNOWN;
+    }
+
+    if (result.get()) {
+      printf("[%s] Node %s has current state %s.\n",
+        get_name(), lifecycle_node, result.get()->current_state.label.c_str());
+      return result.get()->current_state.id;
+    } else {
+      fprintf(stderr, "[%s] Failed to get current state for node %s\n",
+        get_name(), lifecycle_node);
+      return lifecycle_msgs::msg::State::PRIMARY_STATE_UNKNOWN;
+    }
+  }
+
+  bool
+  change_state(std::uint8_t transition, std::chrono::seconds time_out = 3_s)
+  {
+    auto request = std::make_shared<lifecycle_msgs::srv::ChangeState::Request>();
+    request->transition.id = transition;
+
+    if (!client_change_state_->wait_for_service(time_out)) {
+      fprintf(stderr, "Service %s is not available.\n",
+        client_change_state_->get_service_name().c_str());
+      return false;
+    }
+
+    auto result = client_change_state_->async_send_request(request);
+    if (result.wait_for(std::chrono::milliseconds(time_out)) != std::future_status::ready) {
+      fprintf(stderr, "[%s] Failed to change state for node %s. Server timed out.\n",
+        get_name(), lifecycle_node);
+    }
+
+    if (result.get()->success) {
+      printf("[%s] Transition %d successfully triggered.\n",
+        get_name(), static_cast<int>(transition));
+      return true;
+    } else {
+      fprintf(stderr, "[%s] Failed to trigger transition %u\n",
+        get_name(), static_cast<unsigned int>(transition));
+      return false;
+    }
+  }
+
+private:
+  std::shared_ptr<rclcpp::client::Client<lifecycle_msgs::srv::GetState>> client_get_state_;
+  std::shared_ptr<rclcpp::client::Client<lifecycle_msgs::srv::ChangeState>> client_change_state_;
+};
+
+void
+callee_script(std::shared_ptr<LifecycleServiceClient> lc_client)
+{
+  auto sleep_time = 10_s;
+
+  {  // configure
+    lc_client->change_state(lifecycle_msgs::msg::Transition::TRANSITION_CONFIGURE);
```

Comment:
Should you be checking the returned value of this function to ensure it changed states?